Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -2819,31 +2820,34 @@ - (void)yapDatabaseModified:(NSNotification *)notification
                             shouldAnimateScrollToBottom = NO;
                         }
                     }
-                    reloadNeighbors(rowChange.newIndexPath);
                     break;
                 }
                 case YapDatabaseViewChangeMove: {
-                    DDLogError(@".... YapDatabaseViewChangeMove: %@", rowChange.collectionKey);
+                    DDLogError(@".... YapDatabaseViewChangeMove: %@, %@, %@",
+                        rowChange.collectionKey,
+                        rowChange.indexPath,
+                        rowChange.newIndexPath);
                     [self.collectionView deleteItemsAtIndexPaths:@[ rowChange.indexPath ]];
                     [self.collectionView insertItemsAtIndexPaths:@[ rowChange.newIndexPath ]];
-                    reloadNeighbors(rowChange.indexPath);
-                    reloadNeighbors(rowChange.newIndexPath);
                     break;
                 }
                 case YapDatabaseViewChangeUpdate: {
-                    DDLogError(@".... YapDatabaseViewChangeUpdate: %@", rowChange.collectionKey);
-                    YapCollectionKey *collectionKey = rowChange.collectionKey;
-                    OWSAssert(collectionKey.key.length > 0);
-                    if (collectionKey.key) {
-                        ConversationViewItem *viewItem = self.viewItemMap[collectionKey.key];
-                        [self reloadViewItem:viewItem];
-                    }
+                    DDLogError(
+                        @".... YapDatabaseViewChangeUpdate: %@, %@", rowChange.collectionKey, rowChange.indexPath);
                     [self.collectionView reloadItemsAtIndexPaths:@[ rowChange.indexPath ]];
-                    reloadNeighbors(rowChange.indexPath);
+                    [rowsThatChangedSize removeObject:@(rowChange.indexPath.row)];
                     break;
                 }
             }
         }
+
+        // The changes performed above may affect the size of neighboring cells,
+        // as they may affect which cells show "date" headers or "status" footers.
+        NSMutableArray<NSIndexPath *> *rowsToReload = [NSMutableArray new];
+        for (NSNumber *row in rowsThatChangedSize) {
+            [rowsToReload addObject:[NSIndexPath indexPathForRow:row.integerValue inSection:0]];
+        }
+        [self.collectionView reloadItemsAtIndexPaths:rowsToReload];
```

Comment:
I haven't found a better way to do this ... yet.  But I'm wondering if there is a better way.